File: 0455-assign-cookies/0455-assign-cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        childIndex=0
        CookieIndex=0
        g.sort()
        s.sort()
        while childIndex<len(g) and CookieIndex<len(s):
            if s[CookieIndex]>=g[childIndex]:
                childIndex+=1
            CookieIndex+=1
        return childIndex

File: 0455-assign-cookies/0455-assign-cookies.java
import java.util.Arrays;

class Solution {
    public int findContentChildren(int[] g, int[] s) {
        int childIndex = 0;
        int cookieIndex = 0;
        Arrays.sort(g);
        Arrays.sort(s);
        
        while (childIndex < g.length && cookieIndex < s.length) {
            if (s[cookieIndex] >= g[childIndex]) { 
                childIndex++;  // Move to the next child if current cookie satisfies this child
            }
            cookieIndex++;  // Move to the next cookie
        }
        
        return childIndex;  // childIndex now holds the count of satisfied children
    }
}
